syncany-lib/src/main/java/org/syncany/operations/down/DownOperation.java

```diff
@@ -127,82 +128,98 @@ public DownOperation(Config config, DownOperationOptions options) {
+			transferManager.clearPendingTransactions();
 
-		// If we do down, we are no longer allowed to resume a transaction
-		transferManager.clearResumableTransactions();
+			DatabaseBranch localBranch = localDatabase.getLocalDatabaseBranch();
+			List<DatabaseRemoteFile> newRemoteDatabases = result.getLsRemoteResult().getUnknownRemoteDatabases();
 
-		DatabaseBranch localBranch = localDatabase.getLocalDatabaseBranch();
-		List<DatabaseRemoteFile> newRemoteDatabases = result.getLsRemoteResult().getUnknownRemoteDatabases();
+			SortedMap<File, DatabaseRemoteFile> unknownRemoteDatabasesInCache = downloadUnknownRemoteDatabases(newRemoteDatabases);
+			SortedMap<DatabaseRemoteFile, List<DatabaseVersion>> remoteDatabaseHeaders = readUnknownDatabaseVersionHeaders(unknownRemoteDatabasesInCache);
+			Map<DatabaseVersionHeader, File> databaseVersionLocations = findDatabaseVersionLocations(remoteDatabaseHeaders,
+					unknownRemoteDatabasesInCache);
 
-		SortedMap<File, DatabaseRemoteFile> unknownRemoteDatabasesInCache = downloadUnknownRemoteDatabases(newRemoteDatabases);
-		SortedMap<DatabaseRemoteFile, List<DatabaseVersion>> remoteDatabaseHeaders = readUnknownDatabaseVersionHeaders(unknownRemoteDatabasesInCache);
-		Map<DatabaseVersionHeader, File> databaseVersionLocations = findDatabaseVersionLocations(remoteDatabaseHeaders, unknownRemoteDatabasesInCache);
+			Map<String, CleanupRemoteFile> remoteCleanupFiles = getRemoteCleanupFiles();
+			boolean cleanupOccurred = cleanupOccurred(remoteCleanupFiles);
 
-		Map<String, CleanupRemoteFile> remoteCleanupFiles = getRemoteCleanupFiles();
-		boolean cleanupOccurred = cleanupOccurred(remoteCleanupFiles);
+			List<PartialFileHistory> preDeleteFileHistoriesWithLastVersion = null;
 
-		List<PartialFileHistory> preDeleteFileHistoriesWithLastVersion = null;
+			if (cleanupOccurred) {
+				logger.log(Level.INFO, "Cleanup occurred. Capturing local file histories, then deleting entire database ...");
 
-		if (cleanupOccurred) {
-			logger.log(Level.INFO, "Cleanup occurred. Capturing local file histories, then deleting entire database ...");
+				// Capture file histories
+				preDeleteFileHistoriesWithLastVersion = localDatabase.getFileHistoriesWithLastVersion();
 
-			// Capture file histories
-			preDeleteFileHistoriesWithLastVersion = localDatabase.getFileHistoriesWithLastVersion();
+				// Get rid of local database
+				localDatabase.deleteAll();
 
-			// Get rid of local database
-			localDatabase.deleteAll();
+				// Normally, we wouldn't want to commit in the middle of an operation, but unfortunately
+				// we have to, since not committing causes hanging in database operations, since UNCOMMITTED_READ
+				// does not do enough magic to proceed. The commit in itself is not a problem, since we need
+				// to redownload all remote data anyway.
+				try {
+					localDatabase.commit();
+				}
+				catch (SQLException e) {
+					throw new StorageException("Commit failed", e);
```

Comment:
All we do is throw a different exception, and we're either catching or ignore-ing that exception anyway.